core/types/subtyping.cc

```diff
@@ -1259,29 +1259,32 @@ bool isSubTypeUnderConstraintSingle(const GlobalState &gs, TypeConstraint &const
             if (!doesMemberMatch) {
                 result = false;
                 if constexpr (shouldAddErrorDetails) {
-                    string variance;
-                    string joiningText;
-                    switch (idxTypeMember.data(gs)->variance()) {
-                        case Variance::CoVariant: {
-                            variance = "covariant";
-                            joiningText = "a subtype of";
-                            break;
-                        }
-                        case Variance::Invariant: {
-                            variance = "invariant";
-                            joiningText = "equivalent to";
-                            break;
-                        }
-                        case Variance::ContraVariant: {
-                            variance = "contravariant";
-                            joiningText = "a supertype of";
-                            break;
+                    if (!(a2->klass == Symbols::Hash() &&
+                          idxTypeMember.data(gs)->name == core::Names::Constants::Elem())) {
```

Comment:
I think you can flip the condition and then do a early `break` here, to avoid the additional nesting.